python/cudf/cudf/core/single_column_frame.py

```diff
@@ -106,6 +108,19 @@ def _column(self) -> ColumnBase:
     @property  # type: ignore
     @_performance_tracking
     def values(self) -> cupy.ndarray:
+        if (
+            self.ndim == 1
+            and not self._column.has_nulls()
+            and not isinstance(
+                self._column.dtype,
+                np.dtypes.DateTime64DType
+                | np.dtypes.TimeDelta64DType
+                | cudf.CategoricalDtype
+                | object,
+            )
+            and hasattr(self._column, "__cuda_array_interface__")
```

Comment:
Can you ensure `hasattr(self._column, "__cuda_array_interface__")` is not triggering a copy when copy-on-write is enabled?

Reply:
Thanks, I think we don't need it.